helpers/response/query_response.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime
+from graphql import GraphQLError
+
+
+def filter_rooms_by_responses(
+    Query, info, upper_limit_count, lower_limit_count
+):
+    all_responses = Query().get_all_reponses(info)
+    filtered_responses = []
+    for response in all_responses:
+        reponse_count = response.total_responses
+        if lower_limit_count <= reponse_count <= upper_limit_count:
+            filtered_responses.append(response)
+    return filtered_responses
+
+
+def filter_response_by_date(Query, info, upper_date_limit, lower_date_limit):
+    all_responses = Query().get_all_reponses(info)
+    filtered_responses = []
+    for responses in all_responses:
+        for response in responses.response:
+            check_date_limits(
+                response,
+                upper_date_limit, lower_date_limit, filtered_responses
+            )
+        setattr(responses, 'response', filtered_responses)
+    return all_responses
+
+
+def check_date_limits(
+    response, upper_date_limit, lower_date_limit, filtered_responses
```

Comment:
This function should also validate against current date. We have a validation for dates in `helpers/calendar/analytics_helper.py` try to see how it works. the validation is in the function below
```
def validate_current_date(self, start_date, end_date):
        """
        Checks for today's date
        """
        start_date, end_date = CommonAnalytics.convert_dates(
            self, start_date, end_date)

        now = datetime.utcnow().strftime('%b %d %Y')
        date_now = (datetime.strptime(now, "%b %d %Y") + relativedelta(
            days=1)).isoformat() + 'Z'
        user_start_date = dateutil.parser.parse(start_date)
        user_end_date = dateutil.parser.parse(end_date)
        today = dateutil.parser.parse(date_now)
        if user_start_date > today or user_end_date > today:
            raise GraphQLError(
                "Invalid date. You can not retrieve data beyond today")
        return(start_date, end_date)
```

Reply:
Thank you @michael-basweti  for pointing this out. Currently working on this.